Add datetime lookup plugin

willthames opened the pull request:
##### SUMMARY
If you want to access date time information at any time after
gathering facts, and without running setup, a datetime lookup
plugin would be handy
##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
datetime

willthames committed:
Add datetime lookup plugin

If you want to access date time information at any time after
gathering facts, and without running setup, a datetime lookup
plugin would be handy

Teebor-Choka commented on lib/ansible/plugins/lookup/datetime.py:
```diff
@@ -0,0 +1,87 @@
+
+- name: gather the time at the start of an operation
+  set_fact:
+    end_time: "{{ lookup('datetime') }}"
```
The task name is incorrect (copy-paste error). Please, consider introducing a proper task name:

```yaml
- name: gather the time at the end of the operation
```

willthames replied:
@Teebor-Choka fixed, thank you.

willthames committed:
Add datetime lookup plugin

If you want to access date time information at any time after
gathering facts, and without running setup, a datetime lookup
plugin would be handy

bcoca commented:
Hi!

Thank you very much for your submission to Ansible. It sincerely means a lot to us that you've taken the time to contribute.

Per recent announcements at AnsibleFest Atlanta 2019, going forward we are no longer accepting new one off modules/plugins, or modules/plugins that create a new category.

However, we recommend looking into providing this functionality through Ansible Galaxy via Ansible Collections. You can find more information about collections at https://docs.ansible.co
m/ansible/devel/dev_guide/developing_collections.html.

Thank you once again for this and your interest in Ansible!

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

sivel commented:
See also: https://docs.ansible.com/ansible/latest/user_guide/playbooks_templating.html#get-the-current-time
